ledger/daml-on-sql/README.rst

```diff
@@ -0,0 +1,703 @@
+*****************************
+
+Processes and components
+========================
+
+The core processes necessary to run a DAML-on-SQL deployment are:
+
+- the JVM-hosted standalone Ledger API server and
+- the PostgreSQL server used to persist the ledger data.
+
+The Ledger API server communicates with the external world via the gRPC
+Ledger API and communicates with PostgreSQL via JDBC to persist transactions,
+keep track of active contracts, store compiled DAML packages, and so on.
+
+Core architecture considerations
+================================
+
+A very important point to make is that the backing PostgreSQL server performs a
+lot of work which is both CPU- and IO-intensive: all (valid) Ledger API requests
+will eventually hit the database. At the same time, the Ledger API server has to
+have available resources to validate requests, evaluate commands and prepare responses.
+While the PostgreSQL schema is designed to be as efficient as possible, practical
+experience has shown that having **dedicated computation and memory resources for the
+two core components** (the Ledger API server and the PostgreSQL server) allows the two
+to run interfering as little as possible with each other. Depending on the kind of
+deployment you wish to make, this can be achieved with containerization, virtualization
+or simply using physically different machines. Still, the Ledger API communicates
+abundantly with the PostgreSQL server and many Ledger API requests need to go all
+the way to persist information on the database. To reduce the latency necessary to
+serve outstanding requests, **the Ledger API server and PostgreSQL server should be
+physically co-located**.
+
+Core availability considerations
+================================
+
+In order to address availability concerns, it's important to understand what each of the
+core components do and how they interact with each other, in particular regarding state
+and consistency.
+
+In DAML-on-SQL, the Ledger API server holds important state that affects the offset
+assigned to each transaction. This effectively means that having two Ledger API servers
+running on top of a single PostgreSQL server can lead to duplicated and inconsistent
+offsets, effectively leading to undefined (and likely broken) behavior. For this reason,
+it's important to maintain a strict 1:1 relationship between a running Ledger API server
+and a running PostgreSQL server. Note that using PostgreSQL in a high-availability
+configuration does not allow you to run additional Ledger API servers.
+
+Downtime for the Ledger API server can be minimized using a watchdog or orchestration
+system taking care of evaluating its health of the core components and ensuring its
+availability. The Ledger API implementation of DAML-on-SQL exposes the standard gRPC
+health checkpoint that can be used to evaluate the health status of the Ledger API
+component. More information on the endpoint can be found at the documentation for gRPC.
```

Comment:
Putting a link here would be very helpful.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/6850/commits/f57199d0811464e1f0c13d7b9bd3a98f5873cd7a